examples/Iris/step3_optimize.cc

```diff
@@ -0,0 +1,435 @@
+              bool nolegend,
+              size_t image_size,
+              double point_size,
+              T&& model_visualizer = T(),
+              const std::string& output_format = "svg",
+              const std::string& content_type = current::net::constants::kDefaultSVGContentType) {
+  try {
+    const auto ix = features.at(x);
+    const auto iy = features.at(y);
+    const double IrisFlower::*px = ix.mem_ptr;
+    const double IrisFlower::*py = iy.mem_ptr;
+    using namespace current::gnuplot;
+    GNUPlot plot;
+    if (nolegend) {
+      plot.NoBorder().NoTics().NoKey();
+    } else {
+      plot.Title("Iris Data").Grid("back").XLabel(features.at(x).name).YLabel(features.at(y).name);
+    }
+    plot.ImageSize(image_size).OutputFormat(output_format);
+    for (const auto& label : labels) {
+      auto plot_data = WithMeta([&label, &flowers, px, py](Plotter& p) {
+        for (const auto& flower : flowers) {
+          if (flower.Label == label.name) {
+            p(flower.*px, flower.*py);
+          }
+        }
+      });
+      plot.Plot(plot_data.AsPoints()
+                    .Name(label.name)
+                    .Color(label.color)
+                    .Extra("pt " + label.pt + (point_size ? "ps " + current::ToString(point_size) : "")));
+    }
+    model_visualizer.AddPlots(plot, ix.dimension_index, iy.dimension_index);
+    return Response(static_cast<std::string>(plot), HTTPResponseCode.OK, content_type);
+  } catch (const std::out_of_range&) {
+    return Response("Invalid dimension.", HTTPResponseCode.BadRequest);
+  }
+}
+
+enum class ComputationType { TrainDescriptiveModel, TrainDiscriminantModel, ComputeAccuracy };
+
+// The four-dimensional Gaussian with all radii of one has the normalization coefficient of `pi^2`:
+//
+// * the definite integral of `exp(-(x*x + y*y + z*z + t*t)) / (pi^2)` is ``1.
+// * Ref.: http://www.wolframalpha.com/input/?i=exp(-(x*x%2By*y%2Bz*z%2Bt*t))%2F(pi^2)
+//
+// The four-dimensional Gaussian with the radii being `r` has the normalization coeffifient of `pi^2 * r^4)`.
+// * Same integral is `1` for `exp(-((x/2)*(x/2) + (y/2)*(y/2) + (z/2)*(z/2) + (t/2)*(t/2))) / (pi^2 * 16)`.
+// * Ref.: http://www.wolframalpha.com/input/?i=exp(-((x/2)*(x/2)%2B(y/2)*(y/2)%2B(z/2)*(z/2)%2B(t/2)*(t/2)))/(pi^2*16)
+//
+// For all the radii being `r`, I move the coefficient under `exp(-(...)^2)`:
+// * Same integral is `1` for `exp(-(x*x+y*y+z*z+t*t) / (r^2)) / (pi^2*(r^4))`.
+// * Ref.: http://www.wolframalpha.com/input/?i=exp(-(x*x%2By*y%2Bz*z%2Bt*t)+%2F+(2^2))+%2F+(pi^2*(2^4))
+//         (Replace `2` by, say, `5`, i.e. { `2^2`, `2^4` -> `5^2`, `5^4` }, and the integral will stay `1`.
+//
+// Thus, as an implementation detail, model parameters to optimize, the per-coordinate radii, are stored
+// as `-log(sqr(r_i))`. This way, where `x` is the corresponding per-cluster radius-parameter:
+// * The negated exponent argument, which is `1/(r^2)`, is `inv_r_squared`, is `exp(x)`.
```

Comment:
I don't quite get, what "which is `1/(r^2)`, is `inv_r_squared`, is `exp(x)" means exactly.